packages/cross-chain/src/services/providerExecutor.ts

```diff
@@ -123,14 +135,159 @@ class ProviderExecutor {
+     * ```
+     */
+    track(params: {
+        txHash: Hex;
+        providerId: string;
+        originChainId: number;
+        destinationChainId: number;
+        timeout?: number;
+    }): IntentTracker {
+        const tracker = this.getOrCreateTracker(params.providerId);
+
+        const trackingParams: WatchIntentParams = {
+            txHash: params.txHash,
+            originChainId: params.originChainId,
+            destinationChainId: params.destinationChainId,
+            timeout: params.timeout,
+        };
+
+        tracker.startTracking(trackingParams).catch((error) => {
+            console.error("Tracking error:", error);
+        });
+
+        return tracker;
+    }
+
+    /**
+     * Get the current status of an intent without setting up event-based tracking
+     * This is a simple, one-time status check
+     *
+     * @param params - Status query parameters
+     * @returns Current intent status information
+     *
+     * @example
+     * ```typescript
+     * const status = await executor.getIntentStatus({
+     *   txHash: '0x123...',
+     *   providerId: 'across',
+     *   originChainId: 11155111
+     * });
+     *
+     * console.log(status.status); // 'filled' | 'filling' | 'expired'
+     * ```
+     */
+    async getIntentStatus(params: {
+        txHash: Hex;
+        providerId: string;
+        originChainId: number;
+    }): Promise<IntentStatusInfo> {
+        const tracker = this.getOrCreateTracker(params.providerId);
+        return tracker.getIntentStatus(params.txHash, params.originChainId);
+    }
+
+    /**
+     * Get or create a cached tracker for a provider
+     * Uses provider instance's getTrackingConfig() method to get protocol-specific configuration
+     * Trackers are created lazily and cached per provider for efficiency
+     *
+     * @private
+     * @param providerId - Provider ID to get tracker for
+     * @returns IntentTracker instance
+     * @throws {ProviderNotFound} If provider is not registered in executor
+     */
+    private getOrCreateTracker(providerId: string): IntentTracker {
+        if (this.trackerCache.has(providerId)) {
+            return this.trackerCache.get(providerId)!;
+        }
+
+        const provider = this.providers[providerId];
+        if (!provider) {
+            throw new ProviderNotFound(providerId);
+        }
+
+        const config = provider.getTrackingConfig();
+
+        const clientManager = new PublicClientManager(undefined, this.rpcUrls);
+
+        const openWatcher = new OpenEventWatcher({ clientManager });
+
+        const depositInfoParser = new EventBasedDepositInfoParser(config.depositInfoParser, {
+            clientManager,
+        });
+
+        const fillWatcher = new EventBasedFillWatcher(config.fillWatcher, {
+            clientManager,
+        });
+
+        const tracker = new IntentTracker(openWatcher, depositInfoParser, fillWatcher);
+        this.trackerCache.set(providerId, tracker);
+
```

Comment:
Cant we use the factory for this? I think we also could receive initiate the factory with RPC urls and any specific tracking config there and then receive the instance of the factory as a executor dependency in the constructor